### ISSUE TITLE
storage: postgres initial snapshot ingested at a leisurely 30Mb/sec

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#16637]
### What version of Materialize are you using?

environmentd v0.37.0-dev (67362bbf0)

### How did you install Materialize?

Docker image

### What is the issue?

Ingesting 2Gb worth of postgres data takes 70 seconds. No single participating process nor the entire machine is CPU or memory bound, which is underwhelming. It does not feel that the machine is putting up much of an effort. A fully streamlined ingestion should be maxed out along some dimension.

To reproduce:

1. Place the folowing td file in test/pg-cdc:

```
> CREATE SECRET pgpass AS 'postgres'
> CREATE CONNECTION pg FOR POSTGRES
  HOST postgres,
  DATABASE postgres,
  USER postgres,
  PASSWORD SECRET pgpass

$ postgres-execute connection=postgres://postgres:postgres@postgres
ALTER USER postgres WITH replication;

CREATE TABLE t1 (f1 SERIAL PRIMARY KEY, f2 INTEGER DEFAULT 0, f3 TEXT);
ALTER TABLE t1 REPLICA IDENTITY FULL;

CREATE PUBLICATION mz_source FOR ALL TABLES;
INSERT INTO t1 (f3) SELECT REPEAT('a', 1024) FROM generate_series(1, 1024 * 1024);
INSERT INTO t1 (f3) SELECT REPEAT('b', 1024) FROM generate_series(1, 1024 * 1024);

> CREATE SOURCE mz_source
  FROM POSTGRES CONNECTION pg (PUBLICATION 'mz_source')
  FOR ALL TABLES;

> CREATE MATERIALIZED VIEW v1 AS SELECT COUNT(*) FROM t1;
> SELECT * FROM v1;
2097152
```

2. Run it:
```
./mzcompose down -v ; ./mzcompose run default slow.td

```

and observe this:

```
> CREATE SOURCE mz_source FROM POSTGRES CONNECTION pg (PUBLICATION 'mz_source') FOR ALL TABLES;
rows match; continuing at ts 1671005269.6424015
> CREATE MATERIALIZED VIEW v1 AS SELECT COUNT(*) FROM t1;
rows match; continuing at ts 1671005269.9944272
> SELECT * FROM v1;
rows didn't match; sleeping to see if dataflow catches up 50ms 75ms 113ms 169ms 253ms 380ms
rows match; continuing at ts 1671005339.9535296

```
In other words, 70 seconds pass between the time the Pg source is created and the time a very simple view over said source returns the expected result of 2M rows ingested

**Comment by philip-stoev** [comment]
See discussion here https://materializeinc.slack.com/archives/C04R7DV34AY/p1681078734146889

**Comment by petrosagg** [comment]
This got fixed by https://github.com/MaterializeInc/materialize/pull/18055
